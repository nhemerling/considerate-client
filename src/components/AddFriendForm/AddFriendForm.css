.AddFriendForm label {
  font-size: 1.1em;
  display: inline-block;
  margin: 10px 10px 20px;
  margin-left: 0;
}
