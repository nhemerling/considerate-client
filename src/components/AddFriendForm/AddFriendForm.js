import React, { Component } from 'react';
import FriendApiService from '../../services/friend-api-service';

export default class AddFriendForm extends Component {
  static defaultProps = {
    onAddFriendSuccess: () => {},
  };

  state = { error: null };

  handleSubmit = (ev) => {
    ev.preventDefault();
    const { friend_name, occasion, occasion_date } = ev.target;
    const newFriend = {
      friend_name: friend_name.value,
      occasion: occasion.value,
      occasion_date: occasion_date.value,
    };

    this.setState({ error: null });

    FriendApiService.postFriend(newFriend)
      .then((friend) => {
        friend_name.value = '';
        occasion.value = '';
        occasion_date.value = '';
        this.props.onAddFriendSuccess();
      })
      .catch((res) => {
        this.setState({ error: res.error });
      });
  };

  render() {
    const { error } = this.state;
    return (
      <form className="AddFriendForm" onSubmit={this.handleSubmit}>
        <div role="alert">{error && <p className="red">{error}</p>}</div>
        <div className="friend_name">
          <label htmlFor="AddFriendForm__friend_name">Friend name</label>
          <input
            name="friend_name"
            type="text"
            required
            id="AddFriendForm__friend_name"
          ></input>
        </div>
        <div className="occasion">
          <label htmlFor="AddFriendForm__occasion">Occasion</label>
          <input
            name="occasion"
            type="text"
            required
            id="AddFriendForm__occasion"
          ></input>
        </div>
        <div className="occasion_date">
          <label htmlFor="AddFriendForm__occasion_date">Occasion Date</label>
          <input
            name="occasion_date"
            type="date"
            required
            id="AddFriendForm__occasion_date"
          ></input>
        </div>
        <button type="reset">Reset</button>
        <button type="submit">Add</button>
      </form>
    );
  }
}
