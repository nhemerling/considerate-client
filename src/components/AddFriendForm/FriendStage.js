import React, { Component } from 'react';
import FriendApiService from '../../services/friend-api-service';

export default class FriendStage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: null,
      friend_name: { value: '', touched: false },
      occasion: { value: '', touched: false },
      occasion_date: { value: '', touched: false },
    };
  }

  handleSubmit = (ev) => {
    ev.preventDefault();
    const { friend_name, occasion, occasion_date } = this.state;
    const newFriend = {
      friend_name: friend_name.value,
      occasion: occasion.value,
      occasion_date: occasion_date.value,
    };

    this.setState({ error: null });

    FriendApiService.postFriend(newFriend)
      .then((friend) => {
        friend_name.value = '';
        occasion.value = '';
        occasion_date.value = '';
        this.props.onFriendStageSuccess(friend);
      })
      .catch((res) => {
        this.setState({ error: res.error });
      });
  };

  updateFriendName(friend_name) {
    this.setState({
      friend_name: { value: friend_name, touched: true },
    });
  }

  updateOccasion(occasion) {
    this.setState({
      occasion: { value: occasion, touched: true },
    });
  }

  updateOccasionDate(occasion_date) {
    this.setState({
      occasion_date: { value: occasion_date, touched: true },
    });
  }

  render() {
    const { error } = this.state;
    return (
      <form
        className="AddFriendForm__friend-stage"
        onSubmit={this.handleSubmit}
      >
        <div role="alert">{error && <p className="red">{error}</p>}</div>
        <div className="friend_name">
          <label htmlFor="AddFriendForm__friend_name">Friend name</label>
          <input
            name="friend_name"
            type="text"
            required
            id="AddFriendForm__friend_name"
            onChange={(e) => this.updateFriendName(e.target.value)}
          ></input>
        </div>
        <div className="occasion">
          <label htmlFor="AddFriendForm__occasion">Occasion</label>
          <input
            name="occasion"
            type="text"
            required
            id="AddFriendForm__occasion"
            onChange={(e) => this.updateOccasion(e.target.value)}
          ></input>
        </div>
        <div className="occasion_date">
          <label htmlFor="AddFriendForm__occasion_date">Occasion Date</label>
          <input
            name="occasion_date"
            type="date"
            required
            id="AddFriendForm__occasion_date"
            onChange={(e) => this.updateOccasionDate(e.target.value)}
          ></input>
        </div>
        <button type="reset">Reset</button>
        <button type="submit">Add</button>
      </form>
    );
  }
}
