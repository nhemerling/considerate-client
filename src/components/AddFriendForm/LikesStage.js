import React, { Component } from 'react';
import FriendApiService from '../../services/friend-api-service';

export default class LikesStage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: null,
      like1_name: { value: '', touched: false },
      like2_name: { value: '', touched: false },
      like3_name: { value: '', touched: false },
    };
  }

  handleSubmit = (ev) => {
    ev.preventDefault();
    const { like1_name, like2_name, like3_name } = this.state;
    const newLikes = {
      likes: [
        { like_name: like1_name.value },
        { like_name: like2_name.value },
        { like_name: like3_name.value },
      ],
    };
    const friendId = this.props.friend.id;
    this.setState({ error: null });

    FriendApiService.postLikes(newLikes, friendId)
      .then((likes) => {
        like1_name.value = '';
        like2_name.value = '';
        like3_name.value = '';
        this.props.onLikesStageSuccess();
      })
      .catch((res) => {
        this.setState({ error: res.error });
      });
  };

  updateLike1(like1) {
    this.setState({
      like1_name: { value: like1, touched: true },
    });
  }

  updateLike2(like2) {
    this.setState({
      like2_name: { value: like2, touched: true },
    });
  }

  updateLike3(like3) {
    this.setState({
      like3_name: { value: like3, touched: true },
    });
  }

  render() {
    const { error } = this.state;
    return (
      <form className="AddFriendForm__likes-stage" onSubmit={this.handleSubmit}>
        <div role="alert">{error && <p className="red">{error}</p>}</div>
        <div className="friend_likes">
          <div className="friend_like1">
            <label htmlFor="AddFriendForm__like1">Like</label>
            <input
              name="like1"
              type="text"
              id="AddFriendForm__like1"
              onChange={(e) => this.updateLike1(e.target.value)}
            ></input>
          </div>
          <div className="friend_like2">
            <label htmlFor="AddFriendForm__like1">Like</label>
            <input
              name="like2"
              type="text"
              id="AddFriendForm__like2"
              onChange={(e) => this.updateLike2(e.target.value)}
            ></input>
          </div>
          <div className="friend_like3">
            <label htmlFor="AddFriendForm__like1">Like</label>
            <input
              name="like3"
              type="text"
              id="AddFriendForm__like3"
              onChange={(e) => this.updateLike3(e.target.value)}
            ></input>
          </div>
        </div>
        <button type="reset">Reset</button>
        <button type="submit">Add</button>
      </form>
    );
  }
}
