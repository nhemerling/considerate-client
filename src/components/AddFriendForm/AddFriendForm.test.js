import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddFriendForm from './AddFriendForm';

describe('AddFriendForm component', () => {
  it('renders an AddFriendForm', () => {
    const wrapper = shallow(<AddFriendForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
