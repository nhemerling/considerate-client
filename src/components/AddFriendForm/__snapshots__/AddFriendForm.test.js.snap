// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFriendForm component renders an AddFriendForm 1`] = `
<section
  className="AddFriendForm"
>
  <form
    className="AddFriendForm"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <div
      className="friend_name"
    >
      <label
        htmlFor="AddFriendForm__friend_name"
      >
        Friend name
      </label>
      <input
        id="AddFriendForm__friend_name"
        name="friend_name"
        onChange={[Function]}
        required={true}
        type="text"
      />
    </div>
    <div
      className="occasion"
    >
      <label
        htmlFor="AddFriendForm__occasion"
      >
        Occasion
      </label>
      <input
        id="AddFriendForm__occasion"
        name="occasion"
        onChange={[Function]}
        required={true}
        type="text"
      />
    </div>
    <div
      className="occasion_date"
    >
      <label
        htmlFor="AddFriendForm__occasion_date"
      >
        Occasion Date
      </label>
      <input
        id="AddFriendForm__occasion_date"
        name="occasion_date"
        onChange={[Function]}
        required={true}
        type="date"
      />
    </div>
    <div
      className="friend_likes"
    >
      <div
        className="friend_like1"
        key="1"
      >
        <label
          htmlFor="AddFriendForm__like1"
        >
          Like 
          1
        </label>
        <input
          id="AddFriendForm__like1"
          name={1}
          onChange={[Function]}
          required={true}
          type="text"
        />
      </div>
      <button
        onClick={[Function]}
        type="button"
      >
        + Add Like
      </button>
    </div>
    <button
      onClick={[Function]}
      type="reset"
    >
      Reset
    </button>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</section>
`;
